src/plugins/data/public/search/search_source/search_source.ts

```diff
@@ -184,6 +185,34 @@ export class SearchSource {
     return this.parent;
   }
 
+  /**
+   * Run a search using the search service
+   * @return {Promise<SearchResponse<unknown>>}
+   */
+  private runSearch(
+    searchRequest: SearchRequest,
+    esShardTimeout: number,
+    abortSignal?: AbortSignal
+  ) {
+    const abortController = new AbortController();
+    const { signal } = abortController;
+    const searchParams = getSearchParams(getUiSettings(), esShardTimeout);
+    const params = {
+      index: searchRequest.index.title || searchRequest.index,
+      body: searchRequest.body,
+      ...searchParams,
+    };
+    const searching = getSearchService()
+      .search({ params, indexType: searchRequest.indexType }, { signal })
+      .toPromise()
+      .then(({ rawResponse }) => rawResponse);
+
+    const response = searching.then(result => handleResponse(searchRequest, result));
+    if (abortSignal) abortSignal.addEventListener('abort', () => abortController.abort());
+
+    return response;
```

Comment:
Since we're not returning an `abort` method here anymore, I think we can simplify this a bit by removing the creation of a separate `abortController`:

```suggestion
    const searchParams = getSearchParams(getUiSettings(), esShardTimeout);
    const params = {
      index: searchRequest.index.title || searchRequest.index,
      body: searchRequest.body,
      ...searchParams,
    };
    const searching = getSearchService()
      .search({ params, indexType: searchRequest.indexType }, { signal: abortSignal })
      .toPromise()
      .then(({ rawResponse }) => rawResponse);

    return searching.then(result => handleResponse(searchRequest, result));
```